tests/cli.rs

```diff
@@ -75,14 +76,30 @@ fn serves_requests_with_non_default_port(tmpdir: TempDir, port: u16) -> Result<(
     Ok(())
 }
 
-#[rstest]
-fn auth_works(tmpdir: TempDir, port: u16) -> Result<(), Error> {
+#[rstest_parametrize(
+    cli_auth_arg, client_username, client_password,
+    case("testuser:testpassword", "testuser", "testpassword"),
+    case(
+        "testuser:sha256:9f735e0df9a1ddc702bf0a1a7b83033f9f7153a00c29de82cedadc9957289b05",
+        "testuser",
+        "testpassword"
+    ),
+    case(
+        "testuser:sha512:e9e633097ab9ceb3e48ec3f70ee2beba41d05d5420efee5da85f97d97005727587fda33ef4ff2322088f4c79e8133cc9cd9f3512f4d3a303cbdb5bc585415a00",
+        "testuser",
+        "testpassword"
+    ),
+)]
+fn auth_works(cli_auth_arg: &str, client_username: &str, client_password: &str) -> Result<(), Error> {
+    let tmpdir = self::tmpdir();
```

Comment:
You should use the fixtures rather than calling them explicitly.

Reply:
Hmm, how am I going to combine it with `rstest_parametrize`?

Reply:
I just tested it and the naïve way works just fine.

Reply:
Okay? You didn't answer [my question](https://github.com/svenstaro/miniserve/pull/86#discussion_r279147355)

Reply:
I did, I'm saying there is nothing special to do. Naïve way works fine. Example:

```
#[rstest_parametrize(lol, case(3))]
fn serves_requests_with_no_options(tmpdir: TempDir, lol : usize) -> Result<(), Error> {
    let mut child = Command::cargo_bin("miniserve")?
        .arg(tmpdir.path())
        .stdout(Stdio::null())
        .spawn()?;
```

Reply:
To be fair, your first answer is pretty vague.

Anyway, I have done as you requested.